portfolio/src/main/webapp/script.js

```diff
@@ -122,19 +122,23 @@ function createHistoryElement(comment) {
     sentimentButtonElement.style.backgroundColor = "rgb(" + newScore + ", 0, 0)";   
   }
 
-  const deleteButtonElement = document.createElement('button');
-  deleteButtonElement.innerText = 'Delete';
-  deleteButtonElement.className = 'delete-button';
-  deleteButtonElement.addEventListener('click', () => {
-    deleteHistory(comment);
-    historyElement.remove();
- });
 
   historyElement.appendChild(titleElement);
   historyElement.appendChild(sentimentButtonElement);
   historyElement.appendChild(ttBox);
   ttBox.appendChild(ttBoxText);
-  historyElement.appendChild(deleteButtonElement);
+
+  //If the comment is that of the current user, add a delete button
+  if (comment.usercomment == true) {
```

Comment:
Besides this ui hiding delete button, you might also want to add check for user login and is current user in the delete servlet on server.